index.js

```diff
@@ -309,6 +309,14 @@ class ExecutorQueue extends Executor {
             jobId,
             blockedBy: blockedBy.toString()
         }]);
+        const deleteKey = `deleted_${jobId}_${buildId}`;
+
+        // This is to prevent the case where a build is aborted while still in buildQueue
+        // The job might be picked up by the worker, so it's not deleted from buildQueue here
+        // Immediately after, the job gets put back to the queue, so it's always inside buildQueue
+        // This key will be cleaned up automatically or when it's picked up by the worker
+        await this.redisBreaker.runCommand('set', deleteKey, '');
+        await this.redisBreaker.runCommand('expire', deleteKey, 60); // expire in 1 min
```

Comment:
5 mins? not sure if 1 min is too short... Otherwise LGTM 👍

Reply:
sure. I updated to 5 mins. should be more than enough.

Reply:
or maybe we can set to even higher, in case all the workers are somehow are busy for a long time? I can set to 30 mins if we want to be super sure.